diffoscope: 98 -> 99

dtzWill opened the pull request:
Version bump.


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
diffoscope: 98 -> 99

dtzWill commented:
https://salsa.debian.org/reproducible-builds/diffoscope/commit/0cb74c0974bf05534882b23fbddf22b4fc4402c6 I think?

Also:
http://metadata.ftp-master.debian.org/changelogs/main/d/diffoscope/diffoscope_99_changelog

dtzWill committed:
diffoscope: add lz4 tool

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=ac9077f1-e06e-4580-88bf-5f27760c8d4e)

The following builds were skipped because they don't evaluate on x86_64-darwin: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=4dfe0558-b394-4602-b846-292cb17fe520)

Attempted: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```
make: Leaving directory '/build/diffoscope/doc'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99
gzipping man pages under /nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99/share/man/
strip is /nix/store/7iyn7gn33i7xxjgmwf25k20246y6nd9d-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99/lib  /nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99/bin
patching script interpreter paths in /nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99
checking for references to /build in /nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99...
wrapping `/nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99/bin/diffoscope'...
/nix/store/ww2bymwd4j0h514i0dm6bqihd8q75jzc-diffoscope-99
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=9a879658-4486-4298-8fb9-0b2834060b75)

Attempted: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```
make: Leaving directory '/build/diffoscope/doc'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99
gzipping man pages under /nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99/share/man/
strip is /nix/store/90vmpr41dzsx350k5argycaf693hnl1l-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99/lib  /nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99/bin
patching script interpreter paths in /nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99
checking for references to /build in /nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99...
wrapping `/nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99/bin/diffoscope'...
/nix/store/s4h3969y5f4mkl6fw0fz2f0b5wakmk5l-diffoscope-99
```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=95b83005-9599-45fd-a1bf-9e70ca33b4d0)

The following builds were skipped because they don't evaluate on x86_64-darwin: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=c9881370-6da9-4c75-8c8d-9e3916b83f72)

Attempted: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```
make: Leaving directory '/build/diffoscope/doc'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99
gzipping man pages under /nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99/share/man/
strip is /nix/store/7iyn7gn33i7xxjgmwf25k20246y6nd9d-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99/lib  /nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99/bin
patching script interpreter paths in /nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99
checking for references to /build in /nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99...
wrapping `/nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99/bin/diffoscope'...
/nix/store/mx25vpyd77pcix5kzvdhp0ybp1msh2mi-diffoscope-99
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=148049e5-9a83-4217-b3cd-114c51794a75)

Attempted: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```
make: Leaving directory '/build/diffoscope/doc'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99
gzipping man pages under /nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99/share/man/
strip is /nix/store/90vmpr41dzsx350k5argycaf693hnl1l-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99/lib  /nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99/bin
patching script interpreter paths in /nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99
checking for references to /build in /nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99...
wrapping `/nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99/bin/diffoscope'...
/nix/store/497blvagggc10kcs2ighp0f7wrvpjy75-diffoscope-99
```
</p></details>

dtzWill committed:
diffoscope: fix manpage generation

Was already broken, not sure for how long.

Without this the tool fails to run,
so help2man ends up producing an empty file.

dtzWill commented:
> Does pkgs/tools/misc/diffoscope/list-tools.sh list new things to be added as tools?

D'oh i forgot about that.  Was using the resulting tool's "--list-tools".

And.. nope! Same before/after! :)

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=0829ed48-c643-42fa-acf2-a511fe251f60)

The following builds were skipped because they don't evaluate on x86_64-darwin: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=faa7ca66-153e-4d06-843e-83eee45e58c5)

Attempted: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```
make: Leaving directory '/build/diffoscope/doc'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99
gzipping man pages under /nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99/share/man/
strip is /nix/store/7iyn7gn33i7xxjgmwf25k20246y6nd9d-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99/lib  /nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99/bin
patching script interpreter paths in /nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99
checking for references to /build in /nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99...
wrapping `/nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99/bin/diffoscope'...
/nix/store/km7psyq4jl6znbkk8gsgdidfjsaxc4hm-diffoscope-99
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43769&attempt_id=f67e0566-7e2d-4ca3-99b3-b96ac44172ae)

Attempted: diffoscope

<details><summary>Partial log (click to expand)</summary><p>

```
make: Leaving directory '/build/diffoscope/doc'
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99
gzipping man pages under /nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99/share/man/
strip is /nix/store/90vmpr41dzsx350k5argycaf693hnl1l-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99/lib  /nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99/bin
patching script interpreter paths in /nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99
checking for references to /build in /nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99...
wrapping `/nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99/bin/diffoscope'...
/nix/store/3pjrc5ydb14h05vx2i2i3g0b7kzygyxc-diffoscope-99
```
</p></details>

dezgeg commented:
Ok, great. Probably I should document the script so people actually know it exists :).
